hedera-node/docs/design/app/grpc.md

```diff
@@ -12,4 +12,32 @@ rather than the code generated by `protoc`. For these reasons, the built-in gRPC
 It turns out that gRPC is an extremely simple API built on top of HTTP/2, and we can therefore have our own simple gRPC
 framework built on top of an HTTP/2 server (like Netty) and get everything we want.
 
+## Architecture Overview
+
+The gRPC package in Hedera App defines `GrpcServerManager` interface with a number of lifecycle methods such as
+`start()`, `stop()` and `isRunning()`, and an implementation of this interface based on Netty. Hedera App runs at least
+one gRPC server on the port specified in the config, and optionally it will also attempt to run additional gRPC servers
+on the TLS or node operator ports specified in the same config.
+
+### NettyGrpcServerManager
+
+`NettyGrpcServerManager` is an implementation of `GrpcServerManager` and it uses the Hedera App's services registry in
+order to gather the set of RPC endpoints and their corresponding handlers. Given that set and using the
+`GrpcServiceBuilder` class we produce gRPC `ServerServiceDefinition`s which can be directly registered with the Netty
+server.
+
+### GrpcServiceBuilder
```

Comment:
Suggested slight tweak: "As mentioned earlier, the `GrpcServiceBuilder` produces gRPC `ServerServiceDefinitions`. These definitions are configured with the appropriate service handlers and gRPC `ServerCall.Listener `methods. At runtime, the Netty server invokes these listener methods. For instance, the `onMessage()` method is where service logic is integrated to handle transactions or queries using the abstract `MethodBase` type."